### ISSUE TITLE
PheWAS - Populate functional consequence field

### ISSUE THREAD

**Issue by ireneisdoomed** [opentargets/platform#1243]
The `functional_consequence` value for the evidence strings is currently hardcoded to `http://purl.obolibrary.org/obo/SO_0001060`.

The aim is to populate this field using the Sequence Ontology.

**Comment by d0choa** [comment]
Since you're going to do the mappings using the variant index in the genetics portal, it would be great to add consistency to the phewas catalog variant ids.

Currently the variant object inside the phewas catalog evidence looks like the next:

```
  "variant": {
    "type": "snp single",
    "id": "http://identifiers.org/dbsnp/rs6511720"
  },
```

It would be great to migrate to a similar struct to the genetics portal evidence:

```
  "variant": {
    "type": "snp single",
    "rs_id": "rs10793308",
    "id": "11_78374959_C_T"
  },
```
Genetics portal also contain the fields `type_link` and `source_link`. These won't be necessary. 

Hopefully, this won't add extra work and it would be very valuable to make all variant identifiers consistent

**Comment by AsierGonzalez** [comment]
Is there enough information in the [PheWAS file](https://storage.googleapis.com/otar000-evidence_input/PheWAS/data_files/phewas-catalog-19-10-2018.csv) to do that? I don't think you can use the rs id only to get the alleles reliably, can you?

```
"chromosome","basepair","gene","snp","phewas_code","phewas_string","cases","odds_ratio","p"
4,110659067,"*CFI","rs10033900","008","Intestinal infection",406,1.107,0.1561
```

**Comment by ireneisdoomed** [comment]
Changes addressed in the PR [#50](https://github.com/opentargets/evidence_datasource_parsers/pull/50).

Variant index information was gathered and grouped by every `gene, rsid` pair that is present in the PheWAS data.
This was then joined with the PheWAS source file so that we could obtain the `variant.id` term and the `functional_consequence` value. The current `variant` field is now in line with the `Genetics Portal` data, as David mentioned above:

```
 "variant": {
    "type": "snp single",
    "rs_id": "rs10793308",
    "id": "11_78374959_C_T"
  }
```
Please note, for those cases that there are multiple `reference/alternative alleles` for one `rsid`, the `variant.id` field is not present. Likewise, in these cases `variant_id` in `unique_associations_field` is populated with the `rsid`.

Closing issue.

**Comment by d0choa** [comment]
Beautiful work
